packages/react/src/hooks/useStorageURL.ts

```diff
@@ -14,31 +15,32 @@ export interface UseStorageURLResult {
  */
 export const useStorageURL = (
   key: string,
-  options?: S3ProviderGetConfig
-): UseStorageURLResult & { fetch: () => () => void } => {
-  const [result, setResult] = React.useState<UseStorageURLResult>({
-    isLoading: true,
-  });
-
-  // Used to prevent an infinite loop on useEffect, because `options`
-  // will have a different reference on every render
-  const serializedOptions = JSON.stringify(options);
-
-  const fetch = () => {
-    setResult({ isLoading: true });
-
-    const promise = Storage.get(key, options);
-
-    // Attempt to fetch storage object url
-    promise
-      .then((url) => setResult({ url, isLoading: false }))
-      .catch((error: Error) => setResult({ error, isLoading: false }));
+  options?: S3ProviderGetConfig,
+  errorConfig?: UseStorageURLErrorConfig
```

Comment:
This can be done in a follow up PR but imo would be a cleaner API if we extracted the key/values from `errorConfig` and updated the params `useStorageURL` to be an object:
```ts
type UseStorageURLParams {
  key: string;
  fallbackURL?: string;
  onStorageGetError?: (error: Error) => void;
  options?: S3ProviderGetConfig;
}
```

Reply:
Yeah, for sure. This is a miss on me.

Reply:
Addressed in #4253